### DISCUSSION TITLE
Facing issue while running "npm run test"

### DISCUSSION THREAD

**Discussion by Sudarshan-21** [processing/p5.js-web-editor#2804] [Q&A]
I ran "npm run test" and i am gettinf the fails in test suite while all tests are passed.
I have not made any changes to the mentioned  /node_modules/@adobe/css-tools/dist/index.cjs:118  file.
it is something related to jest not able to read the that particular file.
please help me to resolve this.
OS: Linux mint 21.2

![Screenshot from 2023-12-29 23-17-29](https://github.com/processing/p5.js-web-editor/assets/137636943/4dbedbbc-95fd-489f-ab59-a3c87dbf031c)

**Comment by Sudarshan-21** [comment]
@lindapaiste can you please help me out?

**Comment by lindapaiste** [comment]
Hmm this is weird.  I wonder if it's a Linux specific issue.  I would guess that there's nothing specific about that `@adobe/css-tools` file except that it's the first one alphabetically.

I see that the place where it chokes is on a `?.` and that is a new-ish JavaScript feature.  Looks like it's [supported since Node v14](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#browser_compatibility).  I wonder if perhaps you have a lower Node version?  You can try typing the command `node -v` and see what you get.  That is not the whole picture though because you could have multiple Node versions installed.  For example I get `v18.15.0` from the command, but I have my IDE configured to use a `16.x` version for this project.

The other possibility is that there's something in our test configs which doesn't work correctly on Linux.  We should be ignoring CSS files.
https://github.com/processing/p5.js-web-editor/blob/3a80ee689fd6e608e621c57a30a6b45e24ef40d0/package.json#L58-L60

**Reply by lindapaiste** [reply]
> I would guess that there's nothing specific about that `@adobe/css-tools` file except that it's the first one alphabetically.

We can try to rule out the possibility that it's a problem with this particular package.  We want to get rid of that package and see what error you get.  If you get a "test suite failed to run" but with some other package as the source of the error then we know that it's a broader problem.  If the test suite runs (but might fail some snapshot tests) then it is specific to this package.

The `@adobe/css-tools` package is a dependency of `jest-styled-components`.  Try commenting out the line in the `jest.setup.js` file where we load it.
https://github.com/processing/p5.js-web-editor/blob/3a80ee689fd6e608e621c57a30a6b45e24ef40d0/client/jest.setup.js#L2C1-L2C1
And then run the `test` command again.

You may also need to `npm uninstall jest-styled-components`, but probably not.

**Comment by aditya123473892** [comment]
after reinstalling node and NVM it worked for me u can give it a try if nothing works

**Comment by Sudarshan-21** [comment]
@aditya123473892 thank you so much. It worked. Do you know what factor was causing the error?

**Comment by aditya123473892** [comment]
i don't know what caused it did it worked for you

**Reply by Sudarshan-21** [reply]
yes it worked
